ecommerce/hubspot_api.py

```diff
@@ -0,0 +1,135 @@
+"""
+Hubspot Ecommerce Bridge API sync utilities
+
+https://developers.hubspot.com/docs/methods/ecomm-bridge/ecomm-bridge-overview
+"""
+import time
+from urllib.parse import urljoin, urlencode
+import requests
+from django.conf import settings
+
+
+HUBSPOT_API_BASE_URL = "https://api.hubapi.com"
+
+
+def send_hubspot_request(
+    endpoint, api_url, method, body=None, query_params=None, **kwargs
+):
+    """
+    Send a request to Hubspot using the given params, body and api key specified in settings
+
+    Args:
+        endpoint (String): Specific endpoint to hit. Can be the empty string
+        api_url (String): The url path to append endpoint to
+        method (String): GET, POST, or PUT
+        body (serializable data): Data to be JSON serialized and sent with a PUT or POST request
+        query_params (Dict): Params to be added to the query string
+        kwargs: keyword arguments to add to the request method
+    Returns:
+        HTML response to the constructed url
+    """
+
+    base_url = urljoin(HUBSPOT_API_BASE_URL, api_url)
+    if endpoint:
+        base_url = urljoin(base_url, endpoint)
```

Comment:
```python
    base_url = urljoin(f"{HUBSPOT_API_BASE_URL}/", api_url)
    if endpoint:
        base_url = urljoin(f"{base_url}/", endpoint)
```

Reply:
Why?

Reply:
Try running the management command as is: 
```
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://api.hubapi.com/extensions/ecomm/v1/status?hapikey=c8672dc8-a475-4f4d-988a-dfbe65ac099f
```

Reply:
Yes I see.